### PR TITLE
Refactoring logic of delete before [FIND_FULL] time

### PR DESCRIPTION
**Pull request by g0djan**

### CODE REVIEW

File: internal/storages/storage/object.go [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+import (
+	"time"
+)
 
 type Object interface {
 	GetName() string
 	GetLastModified() time.Time
 }
+
+type SortableObjectsSlice []Object
```
Comment (Tinsane) [inline internal/storages/storage/object.go]: https://golang.org/pkg/sort/#Slice

File: cmd/delete.go [hunk 2]
```diff
@@ -117,19 +120,22 @@ func runDeleteBefore(cmd *cobra.Command, args []string) {
...
-		if confirmed {
-			err = storage.DeleteObjectsWhere(folder, func(object storage.Object) bool {
-				return earlierCreated(object, target)
-			})
-		}
-	} else {
-		internal.HandleDeleteBeforeTime(folder, before, modifier, !confirmed)
+		beforeStr = fetchBackupName(potentialTarget)
+	}
+	target, err := internal.FindTargetBeforeName(folder, beforeStr, modifier)
```
Comment (Tinsane) [inline cmd/delete.go]: What about replacing this function with a function FindTargetLSNBeforeName? It seems to me that such a substitution will simplify the code.

Comment (g0djan) [reply cmd/delete.go]: I thought about it, but it is postgres specified logic.

File: internal/storages/storage/object.go (hunk 1 above)
Comment (g0djan) [reply internal/storages/storage/object.go]: Done.

**Comment by x4m** [comment]
Thanks, pushed!

**Pull request merged**